examples/uefi/libvirt.tf

```diff
@@ -0,0 +1,43 @@
+provider "libvirt" {
+    uri = "qemu:///system"
+}
+
+resource "libvirt_network" "dev-cluster" {
+   name = "dev-cluster"
+   domain = "dev-cluster.local"
+   mode = "nat"
+   addresses = ["10.0.200.0/24"]
+}
+
+resource "libvirt_volume" "debian9-base" {
+  name = "debian9-base"
+  source = "http://localhost:8000/debian9-uefi.qcow2"
+}
+
+resource "libvirt_volume" "volume" {
+  name = "vm${count.index}"
+  base_volume_id = "${libvirt_volume.debian9-base.id}"
+  count = 2
+}
+
+resource "libvirt_domain" "domain" {
+  name = "vm${count.index}"
+  memory = "256"
+  vcpu = 1
+  nvram = "/var/lib/libvirt/qemu/nvram/vm${count.index}_VARS.fd"
+  nvram_template = "/srv/provisioning/terraform/debian-stable-uefi_VARS.fd"
```

Comment:
Precisely, the template is also a binary file. It's basically the same snapshot of backing memory for the UEFI firmware that should be copied and used by the new domain after it is defined. 

On the test system I have, I have installed the sample UEFI domain (Debian) through virt-install. A new UEFI backing file for the NVRAM was created with the boot entry for the UEFI-compatible GRUB installed. This is the file I'm using as a template for new installation as the EFI partition will always be in the same location as all new domains are copies of the original installed domain.